server/src/main/java/keywhiz/service/daos/SecretController.java

```diff
@@ -179,9 +197,11 @@ public SecretBuilder withGenerationOptions(Map<String, String> generationOptions
      * @return an instance of the newly created secret.
      */
     public Secret build() {
-      try {
         secretDAO.createSecret(name, encryptedSecret, version, creator, metadata, description, type, generationOptions);
         return transformer.transform(secretDAO.getSecretByNameAndVersion(name, version).get());
+/*
+        // TODO: I don't know if propagating the DataAccessException is the right thing to do.
```

Comment:
1. I'm okay with DataAccessException for now. It is a namespaced RuntimeException. We could unwrap/rewrap the cause, but I can't think of a reason right now. You?
2. If there is a leak, it'd be from exceptions nested in DataAccessException. That'd be hard to test exhaustively, generating SQL errors like constraint violations would be a good sanity check.